optuna/study.py

```diff
@@ -409,17 +414,62 @@ def ask(self) -> trial_module.Trial:
 
                 study.tell(trial, x ** 2)
 
+        Example:
+
+            Passing previously defined distributions to the :func:`~optuna.study.Study.ask`
+            method.
+
+            .. testcode::
+
+                import optuna
+
+
+                study = optuna.create_study()
+
+                # For example, the distributions are previously defined when using `create_trial`.
+                distributions = {
+                    "optimizer": optuna.distributions.CategoricalDistribution(["adam", "sgd"]),
+                    "lr": optuna.distributions.LogUniformDistribution(0.0001, 0.1),
+                }
+                trial = optuna.trial.create_trial(
+                    params={"optimizer": "adam", "lr": 0.0001},
+                    distributions=distributions,
+                    value=0.5,
+                )
+                study.add_trial(trial)
+
+                # You can pass the distributions previously defined.
+                trial = study.ask(fixed_distributions=distributions)
+
+                # `optimizer` and `lr` are already suggested and accessible with `trial.params`.
+                assert "optimizer" in trial.params
+                assert "lr" in trial.params
+
+        Args:
+            fixed_distributions:
+                A dictionary containing the parameter names and parameter's distributions. Each
+                parameter in this dictionary is automatically suggested for the returned trial,
+                even when the suggest method is not explicitly invoked by the user. If this
+                argument is set to :obj:`None`, no parameter is automatically suggested.
```

Comment:
Just curious, but wouldn't it be inconsistent with the arguments of the other methods in Optuna to only suggest to the specified ones when values are specified, and not suggest anything when it is `None`? For example, in `plot_param_importance`, if `params` is specified, it will calculate the importance only for the specified ones, and if not, it will calculate the importance for all parameters.

Reply:
Hmm, do you have any idea of the alternative behavior? Being honest, I'm not sure if we should stick the consistency because: (1) this is not a visualization method and (2) "all parameters" are unknown for this method.

Reply:
> "all parameters" are unknown for this method.

It's true that we can't sample everything in advance, because we shouldn't know the whole search space at all when we ask for a trial. `visualization` and `study` methods are different, so we may not have to worry about that much. Let me resolve this conversation without any changes.